controllers/chat/ChatController.js

```diff
@@ -13,9 +13,133 @@ const BetterSocialConstantListUtils = require('../../services/bettersocial/const
 const {CHANNEL_TYPE} = require('../../helpers/constants');
 const ChatAnonUserInfoFunction = require('../../databases/functions/chatAnonUserInfo');
 const BetterSocialCore = require('../../services/bettersocial');
+const {Op} = require('sequelize');
+const crypto = require('crypto');
 
 const v = new Validator();
 
+const generate_channel_id_for_anon_chat = (owner, member) => {
+  const hash = crypto.createHash('sha256');
+  hash.update(owner, member);
+  return hash.digest('hex');
+};
+
+const is_all_anon_user = async (members) => {
+  // check all members users are anonymous
+  let users = await User.findAll({
+    where: {
+      user_id: {
+        [Op.in]: members
+      },
+      is_anonymous: false
+    }
+  });
+  return users.length === 0 ? true : false;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This function checks if all members of a chat are anonymous. It queries the `User` model for users with IDs in the `members` array who are not anonymous (`is_anonymous: false`). If no such users are found, it returns `true`; otherwise, it returns `false`.
```javascript
const is_all_anon_user = async (members) => {
  let users = await User.findAll({
    where: {
      user_id: {
        [Op.in]: members
      },
      is_anonymous: false
    }
  });
  return users.length === 0 ? true : false;
};
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->